lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -883,34 +888,44 @@ def configure_network(self, vm_obj):
             return
 
         network_devices = list()
-        for network in self.params['networks']:
-            if network:
-                if 'ip' in self.params['networks'][network]:
-                    ip = self.params['networks'][network]['ip']
-                    if ip not in IPNetwork(network):
-                        self.module.fail_json(msg="ip '%s' not in network %s" % (ip, network))
-
-                    ipnet = IPNetwork(network)
-                    self.params['networks'][network]['subnet_mask'] = str(ipnet.netmask)
-
-                if 'network' in self.params['networks'][network]:
-                    if get_obj(self.content, [vim.Network], self.params['networks'][network]['network']) is None:
-                        self.module.fail_json(msg="Network %s doesn't exists" % network)
-                elif 'vlan' in self.params['networks'][network]:
-                    network_name = None
-                    dvps = get_all_objs(self.content, [vim.dvs.DistributedVirtualPortgroup])
-                    for dvp in dvps:
-                        if dvp.config.defaultPortConfig.vlan.vlanId == self.params['networks'][network]['vlan']:
-                            network_name = dvp.config.name
-                            break
-                    if network_name:
-                        self.params['networks'][network]['network'] = network_name
-                    else:
-                        self.module.fail_json(msg="VLAN %(vlan)s doesn't exists" % self.params['networks'][network])
+        for network in self.params.get('networks', list()):
+            if 'ip' in network:
+
+                if 'netmask' in network:
+                    pass
+
+                # TODO: What if netmask and network are provided but conflict ?
```

Comment:
Well, I would rather make the interface more simple and not give the choice.
I think @nerzhul suggested this earlier as well, we both prefer a CIDR over a netmask.